src/zarr/abc/store.py

```diff
@@ -221,6 +221,14 @@ def close(self) -> None:
         self._is_open = False
         pass
 
+    async def _set_dict(self, dict: Mapping[str, Buffer]) -> None:
+        """
+        Insert objects into storage as defined by a prefix: value mapping.
+        """
+        for key, value in dict.items():
+            await self.set(key, value)
+        return None
```

Comment:
IIUC, this is just a helper for our test suite, right? I think I'd favor moving this to either a utility function or a public API (something like `Store.set_many()`). Or perhaps our `set_partial_values` signature needs to evolve to handle this type of thing.

Reply:
right now it's just being used in the test suite, so i'd be happy splitting it off into a private stand-alone function that takes a store. the dict part is just for convenience; what we ultimately will need for proper batching is just a store method that takes `tuple[tuple[key, value], ...]`, which I don't think we have yet?